src/Cli/dotnet/commands/dotnet-project/convert/ProjectConvertCommand.cs

```
@@ -35,13 +37,29 @@ public override int Execute()
             throw new GracefulException(LocalizableStrings.DirectoryAlreadyExists, targetDirectory);
         }
 
+        // Find directives (this can fail, so do this before creating the target directory).
```

This is a stupid example, but if you make .cs (a C# file with no name) then call this on it, I think it would fail? Not sure we really care. More relevant, though, I think it would be good to have some error handling logic including reverting state where relevant.

---

> This is a stupid example, but if you make .cs (a C# file with no name) then call this on it, I think it would fail?

You're right, good catch. It will fail with an error saying the target directory already exists because the target directory will be the current directory. I guess we could add a dedicated error message for this case, but it doesn't seem necessary. Users will be able to work around by specifying `--output`. Although the generated project will also have an empty name, so probably won't work (but users can rename it and then it will work).

> More relevant, though, I think it would be good to have some error handling logic including reverting state where relevant.

That sounds complicated. But I don't think it's needed in normal scenarios. That's why all the checks that can fail are done up front, and only then we do the file modifications (creating target directory, moving files over). If any of the checks fail, we fail early. If the checks succeed but any of those file operations fail for some reason, I'm not sure it would be even possible to revert them (what if the revert fails? how can we revert failed move? the original file might be lost, etc).